### DISCUSSION TITLE
Custom bookmark folders or auto-clone based on tag?

### DISCUSSION THREAD

**Discussion by root-hal9000** [TriliumNext/Trilium#4588] [Q&A] [answered]
Hey everyone, still relatively new, but I want to do something that I couldn't quite get if it's possible from reading the docs.

What I want to do - have a few "folder" notes that automatically gather notes with a certain tag from anywhere else in the tree structure -  or perhaps even more advanced, based on some sort of query (all notes of type X, or containing Y text, etc) - but that's for later. My current need is have a note (prob book view type) that automatically gets a clone of  other notes that have a certain tag cloned under it.

So my first thought was the concept of a bookmark folder as mentioned in the wiki - but somehow being able to make multiples of that for different kinds of things I want "auto-cloned" to being children. For example:
1) I create two root notes, People Of Interest and Places of Interest.
2) Any time I create a note elsewhere in the tree and tag it say #people it gets automatically  cloned as a child  under the  "People of Interest" note, same if I tag #place it gets cloned under "Places of interest"

So it seems like something like the Bookmark Folder concept, but being able to make multiple different types of bookmarked folders that gather different bookmark types so to speak.

But to be honest, I read the wiki page for bookmark folder and I might not be understanding the concept in the first place, because I tried that and it didn't work as I thought it would.  I created a root note with both #bookmarked and #bookmarkFolder, and it shows up on the side bar. But then when I bookmark other notes, they also show up on the sidebar and not under the bookmark folder note. Am I supposed to just manually clone them into the bookmark folder rather than bookmarking the notes themselves? From reading the wiki page, I thought that #bookmarkFolder would automagic this process

Thanks for all your help, I am loving it!

**Comment by root-hal9000** [comment]
Ok, I just read issue #4016  -- so it seems that I am supposedly to manually clone them into the bookmark folder.  so what's the point of the #bookmarkFolder tag itself then? If am manually cloning, I don't see how that tag does something special?

**Reply by rom1dep** [reply]
I'm not sure #bookmarkFolder is what you think it is, as I understand it, it's just a hint so that the UI renders a menu in the bookmark area of the side bar when the associated bookmark carries this tag

**Comment by rom1dep** [comment, marked as answer]
> notes that automatically gather notes with a certain tag from anywhere else in the tree structure - or perhaps even more advanced, based on some sort of query (all notes of type X, or containing Y text, etc)

This can probably be accomplished with a saved search note, just start a search, and click on "Save to note". When accessing this note you'll have the reference of all search results.

**Reply by root-hal9000** [reply]
This definitely solves it for me - and covers the more advanced use case I was thinking with complex queries! Thanks @rom1dep

**Reply by rom1dep** [reply]
Good to hear :)

**Reply by root-hal9000** [reply]
@rom1dep may I ask you a follow up question - since you seem to be an advanced user. So I am creating a saved search to find all incomplete tasks - I am using the built in Task manager template for a TODO by creating notes elsewhere in the tree with the attributes of #task ~template=task_template. Now I want to do a saved search that collects all my incomplete tasks, by searching with the string `#task AND #dateDone=` but I don't know how to essentially say dateDone=NULL. tried dateDone='', dateDone="", dateDone=null, etc ... basically, I want to gather all that don;t have a date done value set. I guess i could just replace with a done boolean, but curious how to do unset attribute values

**Reply by root-hal9000** [reply]
I wrote #dateDone above, but I am using the correct #doneDate for that template. I realized if I set a date and then delete it, #doneDate="" does work! so I guess I am looking for the absent of the tag OR it being and empty string
I suppose I will just change the template to have a boolean instead since I don't care necessarily about when it was completed

**Reply by root-hal9000** [reply]
yea, never mind, a boolean checkbox that has never been touch does not equal false, surprisingly. But I also see i misunderstood the Task manager and the TODO directory already does that. I will study the implementation further. Thanks and sorry to bother @rom1dep - I will post a separate discussion thread later to ask about querying for null values

**Reply by jwhonce** [reply]
@ffcruz85 #!doneDate works as well, https://github.com/zadam/trilium/wiki/Search

**Reply by root-hal9000** [reply]
@jwhonce Ah thanks! I missed that. So the tag is missing if it's not set then. Thanks for the tip. I have thus far managed to break the whole demo TODO thing in the mean time haha....
